src/components/Classes/components/JoinClassModal.js

```diff
@@ -1,89 +1,82 @@
-import React from 'react';
+import React, { useState } from 'react';
 import ReactModal from 'react-modal';
 import { Redirect } from 'react-router-dom';
 import {
   Button, Container, Row, Col, FormGroup, Label, Input,
 } from 'reactstrap';
-import * as fetch from '../../../lib/fetch.js';
+import * as fetch from '../../../lib/fetch';
 
-class JoinClassModal extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      wid: '',
-      disableSubmit: false,
-      error: '',
-      redirect: false,
-    };
-  }
+function JoinClassModal({ props }) {
+  const [wid, setWid] = useState('');
+  const [_disableSubmit, setDisableSubmit] = useState(false);
+  const [error, setError] = useState('');
+  const [redirect, setRedirect] = useState(false);
 
-  closeModal = () => {
-    if (this.props.onClose && {}.toString.call(this.props.onClose) === '[object Function]') {
-      this.props.onClose();
+  const closeModal = () => {
+    const { onClose } = props;
+
+    if (onClose && {}.toString.call(onClose) === '[object Function]') {
+      onClose();
     }
 
-    this.setState({
-      wid: '',
-      error: '',
-      disableSubmit: false,
-    });
+    setWid('');
+    setError('');
+    setDisableSubmit(false);
   };
 
-  badInput = () => {
-    if (!this.state.wid) {
-      this.setState({
-        error: 'Please enter the class code. You should get this from your instructor.',
-      });
+  const badInput = () => {
```

Comment:
this is just a cleaning up nearby code thing, but would you mind renaming this method to `isBadInput`? This makes it more clear that it returns a boolean!